src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -1444,14 +1444,14 @@ FindOrCreateWorkerPool(DistributedExecution *execution, WorkerNode *workerNode)
 	{
 		workerPool = lfirst(workerCell);
 
-		if (WorkerNodeCompare(workerPool->node, workerNode, 0) == 0)
+		if (WorkerNodeCompare(&workerPool->node, workerNode, 0) == 0)
 		{
 			return workerPool;
 		}
 	}
 
 	workerPool = (WorkerPool *) palloc0(sizeof(WorkerPool));
-	workerPool->node = workerNode;
+	memcpy(&workerPool->node, workerNode, sizeof(WorkerNode));
```

Comment:
I'm a little concerned that other (perhaps less frequent) calls to FindWorkerNode are similarly broken and that if we ever add a pointer to WorkerNode we'll need to do so in many places. What do you think about always copying the data structure?

https://github.com/citusdata/citus/pull/new/fix_find_worker_node

Reply:
> I'm a little concerned that other (perhaps less frequent) calls to FindWorkerNode are similarly broken and that if we ever add a pointer to WorkerNode we'll need to copy it in many places. What do you think about always copying the data structure? e.g. #2822

Sounds good. I had checked that there is no other broken call to FindWorkerNode.